include/net/net_context.h

```diff
@@ -283,11 +294,14 @@ struct net_context {
 #endif
 	} options;
 
-	/** Network interface assigned to this context */
-	u8_t iface;
+	/** Protocol (UDP, TCP or IEEE 802.3 protocol value) */
```

Comment:
What is "IEEE 802.3 protocol value"? What if value of "IEEE 802.3 protocol value" clashes with value for UDP or TCP?